support-files/mariadb@.service.in

```diff
@@ -64,19 +64,17 @@ PrivateDevices=true
 # Prevent accessing /home, /root and /run/user
 ProtectHome=true
 
-# Execute pre and post scripts as root, otherwise it does it as User=
-PermissionsStartOnly=true
+# Use an environment file to pass variables _WSREP_START_POSITION
+EnvironmentFile=-@MYSQL_UNIX_DIR@/wsrep-start-position%I
 
 # Perform automatic wsrep recovery. When server is started without wsrep,
 # galera_recovery simply returns an empty string. In any case, however,
 # the script is not expected to return with a non-zero status.
 # It is always safe to unset _WSREP_START_POSITION%I environment variable.
 # Do not panic if galera_recovery script is not available. (MDEV-10538)
-ExecStartPre=/bin/sh -c "systemctl unset-environment _WSREP_START_POSITION%I"
-
 ExecStartPre=/bin/sh -c "[ ! -e @bindir@/galera_recovery ] && VAR= || \
  VAR=`@bindir@/galera_recovery --defaults-file=@sysconf2dir@/my%I.cnf`; [ $? -eq 0 ] \
- && systemctl set-environment _WSREP_START_POSITION%I=$VAR || exit 1"
+ && echo _WSREP_START_POSITION%I=$VAR > @MYSQL_UNIX_DIR@/wsrep-start-position%I || exit 1"
```

Comment:
FYI only: I did remove the galera parts to multiinstance in #510, figured it was easier to test.